cogs/bismuth.py

```diff
@@ -446,6 +447,54 @@ def __init__(self, bot):
             await self.bot.add_reaction(ctx.message, '👎')  # Thumb down
             await self.bot.say("Error {}".format(e))
 
+    @commands.command(name='freebismuth', brief="Register your #Bismuth tweet and get free bismuth", pass_context=True)
+    async def freebismuth(self, ctx, tweet: str):
+        # TODO: too much code in common with operation, factorize somehow.
+        try:
+            amount = float(0)  # amount has to be 0
+            operation = 'twitter'
+            user = User(ctx.message.author.id)
+            user_info = user.info()
+            address = user_info['address']
+
+            #validate tweet url.
+            # TODO: Validate tweet likes and retweets as per freebismuth spec
+            if not validators.url(tweet):
+                print("tweet url error")
+                await self.bot.add_reaction(ctx.message, '😟')
+                await self.bot.say("Link to the tweet does not look ok. Command is `Pawer freebismuth <tweet>`")
+                return
+
+            if user_info and user_info['address']:
+                # User exists and validated the terms, has an address
+                # Make sure balance is enough
+                balance = float(user.balance())
+                tweet_id = tweet.split('/')[-1]  # Extract tweet ID
+                msg = "{} freebismuth, tweet ID is {} ".format(ctx.message.author.display_name, tweet_id)
+                fees = BismuthUtil.fee_for_tx(tweet)
+                print(msg)
+                if balance < amount + 0.01:
```

Comment:
Fee has to account for len of openfield.
I guess it's what fees var above holds, use that instead of fixed 0.01, or use real fee calc from a twitter url.